src/core/SpineGenerator.ts

```diff
@@ -335,34 +338,33 @@ export class MeshGenerator {
       }
     }
 
-    // update sub-mesh
-    mesh.clearSubMesh();
-    const renderer = this._meshRenderer;
-    for (let i = 0, l = subMeshItems.length; i < l; ++i) {
-      const item = subMeshItems[i];
-      const { slotName, blendMode } = item;
-      mesh.addSubMesh(item.subMesh);
+    // update sub primitive
+    _spinePrimitive.clearSubPrimitive();
+    const renderer = this._renderer;
+    for (let i = 0, l = subRenderItems.length; i < l; ++i) {
+      const item = subRenderItems[i];
+      const { slotName, blendMode, texture } = item;
+      _spinePrimitive.addSubPrimitive(item.subPrimitive);
       let material = renderer.getMaterial(i);
       if (!material) {
         material = SpineAnimation.getDefaultMaterial(this._engine);
       }
-      let subMeshTexture = texture.texture;
+      let subTexture = texture.texture;
       if (this.separateSlotTextureMap.has(slotName)) {
-        subMeshTexture = this.separateSlotTextureMap.get(slotName);
+        subTexture = this.separateSlotTextureMap.get(slotName);
       }
-      material.shaderData.setTexture("material_SpineTexture", subMeshTexture);
-      // @ts-ignore
-      material._priority = i;
+      material.shaderData.setTexture("material_SpineTexture", subTexture);
```

Comment:
The number of `material` should be the same as the number of `textures`, not the number of `subRendererElements`